pkg/controllers/dashboard/helm/repo_oci.go

```diff
@@ -56,18 +73,38 @@ func (o *OCIRepohandler) onClusterRepoChange(key string, clusterRepo *catalog.Cl
 	if clusterRepo == nil {
 		return nil, nil
 	}
-
 	// Ignore non OCI ClusterRepos
 	if !registry.IsOCI(clusterRepo.Spec.URL) {
 		return clusterRepo, nil
 	}
 
+	// this is to prevent the handler from making calls when the crd is outdated.
+	updatedRepo, err := o.clusterRepoController.Get(key, metav1.GetOptions{})
+	if err == nil && updatedRepo.ResourceVersion != clusterRepo.ResourceVersion {
+		return clusterRepo, nil
+	}
+
+	if shouldResetRetries(clusterRepo) {
+		clusterRepo.Status.NumberOfRetries = 0
+		clusterRepo.Status.NextRetryAt = metav1.Time{}
+	}
+
+	if !clusterRepo.Status.NextRetryAt.IsZero() && clusterRepo.Status.NextRetryAt.Time.After(timeNow()) {
+		return clusterRepo, nil
+	}
+
+	retryPolicy := getRetryPolicy(clusterRepo)
+
+	if clusterRepo.Status.NumberOfRetries > retryPolicy.MaxRetry {
```

Comment:
Feels like this should result in a `Stalled` condition to be set against the object.

Reply:
I would have to see how the UI will behave if we do that, since right now we only use true or false for all kinds of cluster repos

Reply:
Makes sense, added a log message for this